submit.go

```diff
@@ -460,36 +462,62 @@ func (s *submitServer) saveReport(ctx context.Context, p parsedPayload, reportDi
 		return nil, err
 	}
 
-	if s.ghClient == nil {
-		// we're done here
-		log.Println("GH issue submission disabled")
-		return &resp, nil
+	if err := s.submitGithubIssue(ctx, p, listingURL, &resp); err != nil {
+		return nil, err
 	}
 
-	// submit a github issue
-	ghProj := s.githubProjectMappings[p.AppName]
-	if ghProj == "" {
-		log.Println("Not creating GH issue for unknown app", p.AppName)
-		return &resp, nil
-	}
-	splits := strings.SplitN(ghProj, "/", 2)
-	if len(splits) < 2 {
-		log.Println("Can't create GH issue for invalid repo", ghProj)
+	if err := s.submitSlackNotification(p, listingURL); err != nil {
+		return nil, err
 	}
-	owner, repo := splits[0], splits[1]
 
-	issueReq := buildGithubIssueRequest(p, listingURL)
+	return &resp, nil
+}
 
-	issue, _, err := s.ghClient.Issues.Create(ctx, owner, repo, &issueReq)
-	if err != nil {
-		return nil, err
-	}
+func (s *submitServer) submitGithubIssue(ctx context.Context, p parsedPayload, listingURL string, resp *submitResponse) error {
+	if s.ghClient == nil {
+		log.Println("GH issue submission disabled")
```

Comment:
For the record: generally I prefer an early return to lots of `else` statements (hence `return nil` here), though I'm aware of counterarguments to it.